content/en/docs/Reference/declarative-config.md

```diff
@@ -0,0 +1,636 @@
+OLM will not upgrade an installed bundle if the contents of that bundle are updated in the index.
+
+However, there are some cases where a change in the index metadata is preferred. For example:
+- Channel promotion - if you already released a bundle and later decide that you'd like to add it to another channel, simply add an `olm.channel` property to the `olm.bundle`
+- New upgrade edges - if you release a new 1.2.z (e.g. 1.2.4), but 1.3.0 is already released, you can update the index metadata for 1.3.0 to skip 1.2.4.
+
+OLM highly recommends storing index metadata in source control and treating the source-controlled metadata as the source of truth. Updates to index images should:
+1. Update the source-controlled index directory with a new commit.
+2. Build and push the index image. OLM suggests using a consistent tagging taxonomy (e.g. `:latest` or `:<targetClusterVersion>` so that users can receive updates to an index as they become available. 
+
+### Catalog maintainers
+
+With declarative config, catalog maintainers can focus on operator curation and compatibility.
+Since operator authors have already produced operator-specific indexes for their operators, catalog
+maintainers can build their catalog simply by rendering each operator index into a subdirectory of the
+catalog's root index directory.
+
+#### Example
+
+There are many possible ways to build a catalog, but an extremely simple approach would be to:
+
+1. Maintain a single configuration file containing image references for each operator in the catalog
+   ```yaml
+   name: community-operators
+   repo: quay.io/community-operators/catalog
+   tag: latest
+   references:
+   - name: etcd-operator
+     image: quay.io/etcd-operator/index@sha256:5891b5b522d5df086d0ff0b110fbd9d21bb4fc7163af34d08286a2e846f6be03
+   - name: prometheus-operator
+     image: quay.io/prometheus-operator/index@sha256:e258d248fda94c63753607f7c4494ee0fcbe92f1a76bfdac795c9d84101eb317
+   ```
+
+2. Run a simple script that parses this file and creates a new catalog from its references
+   ```bash
+   name=$(yq eval '.name' catalog.yaml)
+   mkdir "$name" 
+   yq eval '.name + "/" + .references[].name' catalog.yaml | xargs mkdir
+   for l in $(yq e '.name as $catalog | .references[] | .image + "|" + $catalog + "/" + .name + "/index.yaml"' catalog.yaml); do
+     image=$(echo $l | cut -d'|' -f1)
+     file=$(echo $l | cut -d'|' -f2)
+     opm render "$image" > "$file"
+   done
+   opm alpha generate dockerfile "$name"
+   indexImage=$(yq eval '.repo + ":" + .tag' catalog.yaml)
+   docker build -t "$indexImage" -f "$name.Dockerfile" .
+   docker push "$indexImage"
+   ```
+
+### Cluster administrators
+
+For cluster administrators, declarative config-based indexes are largely the same as sqlite-based indexes.
+Both index types serve the exact same GRPC interface required by OLM's on-cluster components.
+
+One benefit of declarative config is that it is possible to build an index entirely with Kubernetes APIs:
```

Comment:
Feels like we've circled back to configmaps. Should we mention the etcd size limits/other caveats to this?

Reply:
That's a great point. I'm tempted to just remove this piece of the cluster admin section entirely. WDYT?